gilmour.go

```diff
@@ -523,16 +536,14 @@ func isExpired(timeout time.Duration, startTime time.Time) bool {
 	return time.Now().After(startTime.Add(timeout))
 }
 
-func try(r RetryConf, fn func() error) (err error) {
+func try(g *Gilmour, fn func() error) (err error) {
 	startTime := time.Now()
 	for {
 		err = fn()
-		if isExpired(r.Timeout, startTime) || !isNetworkError(err) || err == nil {
+		if isExpired(g.getRetryTimeout(), startTime) || !isNetworkError(err) || err == nil {
```

Comment:
This approach is incorrect. RetryTimeout should be passed to the operation and not trapped externally. What you are doing here is basically checking if the timeout was exceeded after the timeout happens.